src/transformers/modeling_utils.py

```diff
@@ -3957,13 +3957,14 @@ def _fix_key(key):
 
         # retrieve unintialized modules and initialize before maybe overriding that with the pretrained weights.
         if _fast_init:
-            if remove_prefix_from_model:
-                _loaded_keys = [f"{prefix}.{k}" for k in loaded_keys]
-            elif add_prefix_to_model:
-                _loaded_keys = [k[len(prefix) + 1 :] for k in loaded_keys]
-            else:
-                _loaded_keys = loaded_keys
-            set_initialized_submodules(model, _loaded_keys)
+            if not ignore_mismatched_sizes:
```

Comment:
If `ignore_mismatched_sizes` is True, it means there is some shape mismatch of weight(s) between a model and a checkpoint.

Currently, in this case, those model weights won't be initialized by `_init_weights`:
  - because we defaults to `_fast_init` in `from_pretrained`
  - and this looks the weight names in the checkpoint, marked them as already initialized, and won't apply `_init_weights` to those weights.
  - The above action doesn't take into account if there is a shape mismatch (as this information is done after where we deal with `_fast_init`)

In order to fix the issue, this PR just not to use `_fast_init` if `ignore_mismatched_sizes` is `True`.